cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/AdapterService.java

```diff
@@ -219,86 +213,101 @@ public void removeAdapter(String namespace, String adapterName) throws AdapterNo
                         AdapterSpecification adapterSpec) {
-    ProgramType programType = adapterTypeInfo.getProgramType();
-    if (programType.equals(ProgramType.WORKFLOW)) {
-      Map<String, WorkflowSpecification> workflowSpecs = spec.getWorkflows();
-      for (Map.Entry<String, WorkflowSpecification> entry : workflowSpecs.entrySet()) {
-        Id.Program programId = Id.Program.from(namespaceId, spec.getName(), entry.getValue().getName());
-        addSchedule(programId, adapterSpec);
-      }
-    } else {
-      // Only Workflows are supported to be scheduled in the current implementation
-      throw new UnsupportedOperationException(String.format("Unsupported program type %s for adapter",
-                                                            programType.toString()));
+    checkWorkflow(adapterTypeInfo.getProgramType());
+    Map<String, WorkflowSpecification> workflowSpecs = spec.getWorkflows();
+    for (Map.Entry<String, WorkflowSpecification> entry : workflowSpecs.entrySet()) {
+      Id.Program programId = Id.Program.from(namespaceId, spec.getName(), entry.getValue().getName());
+      addSchedule(programId, SchedulableProgramType.WORKFLOW, adapterSpec);
     }
   }
 
   // Unschedule all the programs needed for the adapter. Currently, only unscheduling of workflow is supported.
   private void unschedule(String namespaceId, ApplicationSpecification spec, AdapterTypeInfo adapterTypeInfo,
                           AdapterSpecification adapterSpec) {
-    ProgramType programType = adapterTypeInfo.getProgramType();
-    if (programType.equals(ProgramType.WORKFLOW)) {
-      Map<String, WorkflowSpecification> workflowSpecs = spec.getWorkflows();
-      for (Map.Entry<String, WorkflowSpecification> entry : workflowSpecs.entrySet()) {
-        Id.Program programId = Id.Program.from(namespaceId, adapterSpec.getType(), entry.getValue().getName());
-        deleteSchedule(programId, SchedulableProgramType.WORKFLOW,
-                       constructScheduleName(programId, adapterSpec.getName()));
-      }
-    } else {
-      // Only Workflows are supported to be unscheduled in the current implementation
-      throw new UnsupportedOperationException(String.format("Unsupported program type %s for adapter",
-                                                            programType.toString()));
+    checkWorkflow(adapterTypeInfo.getProgramType());
+    Map<String, WorkflowSpecification> workflowSpecs = spec.getWorkflows();
+    for (Map.Entry<String, WorkflowSpecification> entry : workflowSpecs.entrySet()) {
+      Id.Program programId = Id.Program.from(namespaceId, adapterSpec.getType(), entry.getValue().getName());
+      deleteSchedule(programId, SchedulableProgramType.WORKFLOW,
+                     constructScheduleName(programId, adapterSpec.getName()));
     }
   }
 
+  private void checkWorkflow(ProgramType programType) {
```

Comment:
This precondition was in four different places, so i figure the error string should just be in one spot.